CMakeLists.txt

```diff
@@ -36,6 +49,15 @@ if(TBB_FOUND)
     "[${TBB_LIBRARY}][${TBB_INCLUDE_DIR}]")
 endif()
 
+# go find nanobind
+execute_process(
+  COMMAND ${Python_EXECUTABLE} "-m" "nanobind" "--cmake_dir"
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+  OUTPUT_VARIABLE NB_DIR)
+list(APPEND CMAKE_PREFIX_PATH "${NB_DIR}")
+message(STATUS ${CMAKE_PREFIX_PATH})
+find_package(nanobind CONFIG REQUIRED)
```

Comment:
```suggestion
# go find nanobind
execute_process(
  COMMAND ${Python_EXECUTABLE} "-m" "nanobind" "--cmake_dir"
  OUTPUT_STRIP_TRAILING_WHITESPACE
  OUTPUT_VARIABLE nanobind_ROOT)
find_package(nanobind CONFIG REQUIRED)
```
The goal here is to find the **specific** nanobind associated with the already found Python. Use the project-specific root variable to accomplish this instead of the global `CMAKE_PREFIX_PATH`.